homeassistant/components/keenetic_ndms2/config_flow.py

```diff
@@ -150,19 +150,26 @@ async def async_step_init(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Manage the options."""
-        router: KeeneticRouter = self.hass.data[DOMAIN][self.config_entry.entry_id][
-            ROUTER
-        ]
-
-        interfaces: list[InterfaceInfo] = await self.hass.async_add_executor_job(
-            router.client.get_interfaces
-        )
+        try:
+            router: KeeneticRouter = self.hass.data[DOMAIN][self.config_entry.entry_id][
+                ROUTER
+            ]
+        except KeyError:
+            return self.async_abort(reason="not_initialized")
```

Comment:
Instead let's check if the value we want to use is there instead of letting it raise a KeyError

Reply:
@joostlek , thanks for the review. done